### DISCUSSION TITLE
How much divergence is there from discord.py?

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [Pycord-Development/pycord#2430] [Q&A] [answered]
Once discord.py became archived, I quickly migrated over to pycord (no regrets). It seemed easier to do some things in pycord versus the old discord.py. For example slash commands were built in, while in discord.py I had to use an extension if I remember correctly.

Anyway, it's been a while since I've given any attention to my discord bot, and I am trying to find a good way to unit test it... but I only see these two frameworks.

- https://github.com/CraftSpider/dpytest
- https://github.com/KoalaBotUK/testcord (unmaintained and outdated fork of dpytest)

I tried testcord, but many things are not working... and there is no support for slash commands. I'm wondering, if pycord is close enough to the un-archived discord.py, where I might be able to use dpytest?

Otherwise, can anyone provide some examples for how to unit test a discord bot? My bot is not too complex. There are a few slash commands, a few views, and 1 modal.

**Comment by MattyTheHacker** [comment]
I can't add much here but I do know unit testing was being worked on to build into pycord, see [1677](https://github.com/Pycord-Development/pycord/pull/1677/)

**Reply by ReenigneArcher** [reply]
I was following that PR and saw it got closed recently... I don't really understand why it got closed, but I guess after 2 years it's unlikely to be easy to integrate.

**Reply by Lulalaby** [reply]
There were too many conflicts, we'd have to redo it from scratch

**Reply by Paillat-dev** [reply]
Can I open a feat request for that ? So at least it is tracked somewhere.

**Reply by MattyTheHacker** [reply, marked as answer]
I have now opened a feature request for this: https://github.com/Pycord-Development/pycord/issues/2813
